File: backend/src/main/java/com/ecommerce/app/catalog/application/mapper/CategoriaMapper.java
package com.ecommerce.app.catalog.application.mapper;

import com.ecommerce.app.catalog.application.dto.request.ActualizarCategoriaRequest;
import com.ecommerce.app.catalog.application.dto.request.CrearCategoriaRequest;
import com.ecommerce.app.catalog.application.dto.response.CategoriaResponse;
import com.ecommerce.app.catalog.domain.model.Categoria;
import org.springframework.stereotype.Component;

@Component
public class CategoriaMapper {

    public Categoria toEntity(CrearCategoriaRequest request) {
        if (request == null) {
            return null;
        }

        Categoria categoria = new Categoria();
        categoria.setNombre(request.nombre());
        return categoria;
    }

    public void updateEntity(Categoria categoria, ActualizarCategoriaRequest request) {
        if (categoria == null || request == null) {
            return;
        }

        categoria.setNombre(request.nombre());
    }

    public CategoriaResponse toResponse(Categoria categoria) {
        if (categoria == null) {
            return null;
        }

        return new CategoriaResponse(
                categoria.getId(),
                categoria.getNombre()
        );
    }
}

File: frontend/src/app/features/catalog/infrastructure/mappers/categoria.mapper.ts
import { Categoria } from '../../domain/models/categoria.model';
import { CategoriaResponseDto } from '../dtos/categoria.dto';

export class CategoriaMapper {
  public static toDomain(dto: CategoriaResponseDto): Categoria {
    return new Categoria(dto.id, dto.nombre);
  }

  public static toDomainList(dtos: CategoriaResponseDto[]): Categoria[] {
    return dtos.map(dto => this.toDomain(dto));
  }
}
